dex-jmh/src/test/scala/com/wavesplatform/dex/model/address/AddressActorStartingBenchmark.scala

```diff
@@ -0,0 +1,177 @@
+        orderPartiallyFilledAndThenCancelledCount
+
+    val maxPrice: Long      = 1000L * Order.PriceConstant
+    val minPrice: Long      = 1L * Order.PriceConstant
+    val priceGen: Gen[Long] = Gen.chooseNum(minPrice, maxPrice)
+
+    val owner: KeyPair = clientGen.sample.get
+
+    val askGen: Gen[Order] = orderGen(owner, priceGen, OrderType.SELL)
+    val bidGen: Gen[Order] = orderGen(owner, priceGen, OrderType.BUY)
+
+    val orders: List[AcceptedOrder] = ordersGen(ordersCount).sample.get
+
+    val (eventsGroups, _) =
+      Seq(
+        orderAddedEventCount                      -> OrderEvent.Added,
+        orderCancelledEventCount                  -> OrderEvent.Cancelled,
+        orderFilledImmediatelyCount               -> OrderEvent.FilledEventually,
+        orderFilledEventuallyCount                -> OrderEvent.FilledEventually,
+        orderPartiallyFilledCount                 -> OrderEvent.PartiallyFilled,
+        orderPartiallyFilledAndThenCancelledCount -> OrderEvent.PartiallyFilledAndThenCancelled
+      ).foldLeft((Seq.empty[Seq[Events.Event]], orders)) { case ((result, ordersRemaining), (groupSize, event)) =>
+        val (ordersGroup, newRemaining) = ordersRemaining.splitAt(groupSize)
+        val eventsGroup =
+          event match {
+            case OrderEvent.Added     => ordersGroup.map(getOrderAddedEvent)
+            case OrderEvent.Cancelled => ordersGroup.flatMap { ao => Seq(getOrderAddedEvent(ao), getOrderCancelledEvent(ao)) }
+            case OrderEvent.FilledImmediately =>
+              ordersGroup.flatMap { ao => Seq(getOrderAddedEvent(ao), getOrderExecutedEvent(ao, getCounter(ao, ao.amount))) }
+            case OrderEvent.FilledEventually =>
+              ordersGroup.flatMap { ao =>
+                val firstExecutedEvent  = getOrderExecutedEvent(ao, getCounter(ao, ao.amount / 2))
+                val secondExecutedEvent = getOrderExecutedEvent(firstExecutedEvent.submittedRemaining, getCounter(ao, ao.amount))
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, secondExecutedEvent)
+              }
+            case OrderEvent.PartiallyFilled =>
+              ordersGroup.flatMap { ao =>
+                val firstExecutedEvent  = getOrderExecutedEvent(ao, getCounter(ao, ao.amount / 2))
+                val secondExecutedEvent = getOrderExecutedEvent(firstExecutedEvent.submittedRemaining, getCounter(ao, ao.amount / 3))
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, secondExecutedEvent)
+              }
+            case OrderEvent.PartiallyFilledAndThenCancelled =>
+              ordersGroup.flatMap { ao =>
+                val firstExecutedEvent = getOrderExecutedEvent(ao, getCounter(ao, ao.amount / 2))
+                val cancelEvent        = getOrderCancelledEvent(firstExecutedEvent.submittedRemaining)
+                Seq(getOrderAddedEvent(ao), firstExecutedEvent, cancelEvent)
+              }
+          }
+        (result :+ eventsGroup) -> newRemaining
+      }
```

Comment:
I think it is easier to use `Gen.frequency` and create a separate generator for each group, then you won't need a predefined orders, e.g.:
```scala
  val addedGen: Gen[Event] = ???

  Gen.frequency(
    150 -> addedGen,
    /* ... */
  )
```
Also this generator create a shuffled result.